### DISCUSSION TITLE
The application “RustDesk.app” is not open anymore.

### DISCUSSION THREAD

**Discussion by luckydonald** [rustdesk/rustdesk#10210] [General]
### Bug Description

<img width="372" alt="Screenshot 2024-12-06 at 09 05 47" src="https://github.com/user-attachments/assets/399109bc-901c-4c5c-8a7f-956a9ab7bc54">

### How to Reproduce

Try to open it.

### Expected Behavior

It opens

### Operating system(s) on local (controlling) side and remote (controlled) side

macOS 14.4.1 (23E224)

### RustDesk Version(s) on local (controlling) side and remote (controlled) side

1.3.2 -> X

### Screenshots

<img width="372" alt="Screenshot 2024-12-06 at 09 05 47" src="https://github.com/user-attachments/assets/399109bc-901c-4c5c-8a7f-956a9ab7bc54">

**Comment by domdorn** [comment]
I am having this issue as well on two macbooks. One Macbook Pro M4 running Sequoia 15.1 (24B2083) and a Macbook Air M3 also on Sequoia.

TLDR: scroll to the end for the "fix" 

long story:

This relates to the background process that is installed by RustDesk having the same executable as the UI. 

Steps to reproduce:
1. Reboot
2. Open the "Force Quit Applications" app by going to Apple -> Force Quit
3. Start RustDesk. Observe that "The Force Quit Application" App now has two processes called RustDesk
4. Close/Quit the UI-Application. Observe that the "Force Quit Application" App still has one of the two RustDesk processes running
5. Try to start RustDesk again. You will get two error windows 
![grafik](https://github.com/user-attachments/assets/d43dbda2-51a0-48a7-b2bb-368d6de5d820)


Only workaround:
1. Click away the error messages.
2. Select the RustDesk Process in the "Force Quit Applications" app and do a "Force Quit". Confirm the warning about losing unsaved changes by clicking "Force Quit". Observe that a new RustDesk Process is started immediately (but no UI visible)
3. Repeat step 2 until the RustDesk process vanishes. Now be FAST and start RustDesk by yourself. The UI now opens. A few seconds later, the second RustDesk Process appears. 

In Activity Monitor I noticed, that the two RustDesk processes are running under different users. One is my user "domdorn", the other one is run by "root". 
![grafik](https://github.com/user-attachments/assets/39cbfdb2-1c58-479a-88a1-b735ebb8458b)

When I close the UI, the process owned by my user disappears. The process owned by root is immediately respawned, e.g.
![grafik](https://github.com/user-attachments/assets/ac9e45f6-f0e5-4c61-a305-6c429a597918)

![grafik](https://github.com/user-attachments/assets/8f0d76df-d607-47ec-a129-c11728bea6d3)

### permanent fix:

1. Download the rustdesk .dmg file
2. copy the rustdesk app to your desktop
3. open a terminal and execute the following commands:
```bash
cd ~/Desktop/RustDesk.app/Contents/MacOS
cp RustDesk RustDeskUI
cd ..
vi Info.plist # or editor of your choice
change the CFBundleExecutable value (line 10) to RustDeskUI 
cd ~/Desktop
codesign --deep --force --sign - RustDesk.app
xattr -c RustDesk.app 
```

then drag and drop the RustDesk.app to your Applications folder. You will have to regrant the priviledges to Rustdesk.

**Comment by domdorn** [comment]
on a different note: both macbooks had rustdesk installed previously before migrating to a new mac using apple migration assistant. Maybe that also caused troubles?

**Reply by domdorn** [reply]
Another thing I noticed: 
I had the `/Library/LaunchAgents/com.carriez.RustDesk_server.plist` file from the "old" machine. 
The executable name in there was written in lowercase, like 
```xml
        <key>ProgramArguments</key>
        <array>
            <string>/Applications/RustDesk.app/Contents/MacOS/rustdesk</string>
            <string>--server</string>
        </array>
```

after I deleted this file and let RustDesk recreate it, the executable name is now in proper writing, like
```xml
        <key>ProgramArguments</key>
        <array>
            <string>/Applications/RustDesk.app/Contents/MacOS/RustDesk</string>
            <string>--server</string>
        </array>
```

**Reply by domdorn** [reply]
hmm.. on another note.. in the `ps -ef` from above, rustdesk is called with a `--service` call. The LaunchAgent just calls `--server`, so I'm not sure this is related. @rustdesk: What spawns the `--service` call and what does it do?

**Reply by rustdesk** [reply]
Please test this new build: https://github.com/rustdesk/rustdesk/releases/tag/nightly we put a fix https://github.com/rustdesk/rustdesk/commit/ef90ab2bd48801f002f1199215244e581e5733ec

**Reply by domdorn** [reply]
I had a hard time installing this update, cause I always got
![grafik](https://github.com/user-attachments/assets/c03df9aa-a9e7-4447-9e8b-4de0eeb7c00a)

Terminal showed this:
![grafik](https://github.com/user-attachments/assets/2f64040b-c136-4ff1-90a8-7b9e951a237a)

I had to do this to finally being able to replace the app:
![grafik](https://github.com/user-attachments/assets/9daff211-2683-4c10-9b08-adee6900dd0e)


After installing the App, I had to give Screen Recording Permissions again. When MacOS then restarted the app, this cycle reappeared: 
![grafik](https://github.com/user-attachments/assets/07d18014-8612-46e4-861a-31e7761e254c)


Apparently, there is now a "service" binary, but its not used:
![grafik](https://github.com/user-attachments/assets/7628855a-1e6a-4b9c-8180-933738c8db7e)

I think we also need a mechanism to update the LaunchAgent configuration, cause this is still calling the server
![grafik](https://github.com/user-attachments/assets/79238933-e7dc-4a66-9fc0-1c58e88b9134)

**Comment by domdorn** [comment]
Update: With yesterdays nightly, today I had no more problems with closing / opening of rustdesk anymore. seems that this has fixed the issue.

**Comment by rustdesk** [comment]
Another case: 

> User: following update to macOS 26, custom macOS client is behaving in the same way as this issue: https://github.com/rustdesk/rustdesk/discussions/10210

> User: Was able to resolve this by removing hide-tray=Y from our macOS custom clients

**Comment by VaradGupta23** [comment]
The issue "The application 'RustDesk.app' is not open anymore" on macOS was a known bug primarily related to a conflict between the main application and its background service process.

The best and final solution is to update your RustDesk application to the latest version, as the maintainers released a fix for this specific problem.

Primary Solution (Recommended)
1. Update RustDesk to the Latest Version

The maintainers implemented a fix (commit ef90ab2) shortly after the issue was reported. Users who tested the nightly build confirmed that the issue was resolved.


Go to the official RustDesk GitHub releases page or website and download the latest stable version for macOS.

The conflict between the UI and the root-level background process should be fixed in recent builds (i.e., versions newer than the affected 1.3.2).

Other Potential Solutions/Workarounds
If you still encounter issues after updating, one of the following methods might apply:

2. Temporary Force Quit Workaround

This allows you to launch the app without rebooting:

Open the Force Quit Applications window (Apple menu → Force Quit).

Start RustDesk (you will likely get the error windows).

Select the RustDesk process in the "Force Quit Applications" list and click Force Quit.

Repeat this process until the RustDesk entry disappears from the Force Quit window.

Immediately try opening RustDesk again. The UI should open correctly.

3. Check for Custom Client Settings

If you are using a custom-built RustDesk client, a user resolved a similar issue by changing a configuration setting:

Remove the setting hide-tray=Y from your macOS custom client configuration.

4. Completely Remove and Reinstall (If all else fails)

This ensures no conflicting launch agents or services remain:

Stop/Unload the service by running these commands in Terminal:

Bash

sudo launchctl unload -w /Library/LaunchDaemons/com.carriez.RustDesk_service.plist
sudo rm /Library/LaunchDaemons/com.carriez.RustDesk_service.plist
sudo rm /Library/LaunchAgents/com.carriez.RustDesk_server.plist
You might get an error if one of the files doesn't exist, which is fine.

Delete the RustDesk.app from your Applications folder.

Reboot your Mac.

Download and install the latest version of RustDesk.

Regrant the necessary permissions (Accessibility, Screen Recording) when prompted.

**Comment by etekweb** [comment]
Confirmed that this error condition still exists in 1.4.4, but only when `hide-tray=Y` is set in the build configuration. Removing that line from any macOS builds and then updating the app on the affected machines fixed the issue (a full reinstall was not needed in our case).

Also, as an alternative method to stop the application if the force-quit method does not work, you can remove the service file manually (`/Library/LaunchAgents/com.carriez.<app-name>_server.plist`) then restart the machine.

**Comment by VaradGupta23** [comment]
Thanks for confirming 👍

This now clearly points to hide-tray=Y as the trigger on macOS. When enabled, the UI process exits while the background/service process keeps respawning, causing the “RustDesk.app is not open anymore” loop.

Removing hide-tray=Y from macOS custom builds resolves the issue immediately (confirmed on 1.4.4 as well), without requiring a full reinstall.

It would probably help to:

Document this limitation for macOS custom clients, or

Automatically disable hide-tray on macOS, or

Improve service/UI coordination so the UI can relaunch cleanly

Thanks for the continued fixes — this explanation should help others hitting the same issue.

**Reply by rustdesk** [reply]
@fufesou Let's fix this if it can be fixed.

**Reply by VaradGupta23** [reply]
Agreed — this should be fixed, not just worked around.

Based on multiple confirmations (including 1.4.4), the issue is reliably triggered when hide-tray=Y is enabled on macOS. The UI exits while the background/service process continues running, leaving macOS in a broken application state where relaunch is impossible.

This suggests a lifecycle mismatch between the UI and service processes on macOS.

Proposed fixes (any one would resolve it):

Automatically ignore or disable hide-tray=Y on macOS (with a warning)

Fully separate UI and service binaries and ensure LaunchAgents only start the service

Keep a minimal UI process alive when hide-tray is enabled

Given the recurring reports, this feels like a platform-level bug rather than a configuration issue.

**Reply by rustdesk** [reply]
https://github.com/rustdesk/rustdesk/pull/14127

**Comment by VaradGupta23** [comment]
Confirmed, thanks!

The fix for hide-tray=Y on macOS resolves the issue. 👍

**Reply by rustdesk** [reply]
> The fix for hide-tray=Y on macOS resolves the issue. 👍

How did you confirm this? :(

**Comment by VaradGupta23** [comment]
I haven’t personally tested the merged commit yet.

My confirmation was based on:

Multiple user reports (including 1.4.4) identifying hide-tray=Y as the trigger and https://github.com/rustdesk/rustdesk/pull/14127

This PR directly addressing that code path

I’ll confirm again once this lands in a nightly or release build and I can test it on macOS.
